lib/widgets/message_list.dart

```diff
@@ -1473,7 +1510,37 @@ class MessageWithPossibleSender extends StatelessWidget {
         child: Icon(ZulipIcons.star_filled, size: 16, color: designVariables.star));
     }
 
-    Widget content = MessageContent(message: message, content: item.content);
+    Widget? revealButton;
+    if (showAsMuted) {
+      revealButton = TextButton.icon(
+        onPressed: () {
+          changeMuteStatus(false);
+        },
+        style: TextButton.styleFrom(
+          minimumSize: Size.zero,
+          padding: EdgeInsets.symmetric(vertical: 4, horizontal: 6),
+          tapTargetSize: MaterialTapTargetSize.shrinkWrap,
+          shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(6)),
+          splashFactory: NoSplash.splashFactory,
+        ).copyWith(
+          backgroundColor: WidgetStateColor.fromMap({
+            WidgetState.pressed: designVariables.neutralButtonBg,
+            ~WidgetState.pressed: designVariables.neutralButtonBg
+              .withFadedAlpha(0),
+          }),
+          foregroundColor: WidgetStateColor.fromMap({
+            WidgetState.pressed: designVariables.neutralButtonLabel,
+            ~WidgetState.pressed: designVariables.neutralButtonLabel
+              .withFadedAlpha(0.85),
+          }),
+        ),
+        icon: Icon(ZulipIcons.eye),
+        label: Text(zulipLocalizations.revealButtonLabel,
+          style: TextStyle(fontSize: 16, height: 1)
+            .merge(weightVariableTextStyle(context, wght: 600))));
+    }
```

Comment:
With 20+ lines of code, this would be good to pull out into a separate widget or a helper method